omnigibson/prims/prim_base.py

```diff
@@ -20,43 +21,42 @@ class BasePrim(Serializable, UniquelyNamed, Recreatable, ABC):
         unless it is a non-root articulation link.
 
     Args:
-        prim_path (str): prim path of the Prim to encapsulate or create.
+        relative_prim_path (str): prim path of the Prim to encapsulate or create.
         name (str): Name for the object. Names need to be unique per scene.
         load_config (None or dict): If specified, should contain keyword-mapped values that are relevant for
             loading this prim at runtime. Note that this is only needed if the prim does not already exist at
-            @prim_path -- it will be ignored if it already exists. Subclasses should define the exact keys expected
+            @relative_prim_path -- it will be ignored if it already exists. Subclasses should define the exact keys expected
             for their class.
     """
 
     def __init__(
         self,
-        prim_path,
+        relative_prim_path,
         name,
         load_config=None,
     ):
-        self._prim_path = prim_path
+        self._relative_prim_path = relative_prim_path
         self._name = name
         self._load_config = dict() if load_config is None else load_config
 
         # Other values that will be filled in at runtime
+        self._scene = None
+        self._scene_assigned = False
         self._applied_visual_material = None
         self._loaded = False  # Whether this prim exists in the stage or not
         self._initialized = False  # Whether this prim has its internal handles / info initialized or not (occurs AFTER and INDEPENDENTLY from loading!)
         self._prim = None
-        self._state_size = None
         self._n_duplicates = 0  # Simple counter for keeping track of duplicates for unique name indexing
 
+        # Check if this prim was created manually. This member will be automatically set for prims
+        # that get created during the _load phase of this class, but sometimes we create prims using
+        # alternative methods and then create this class - in that case too we need to make sure we
+        # add the right xform properties, so callers will just pass in the created manually flag.
```

Comment:
what are the alternative methods?

Reply:
`particle_applier_remover_demo.py` is an example: 
```
    # If we're using a projection volume, we manually add in the required metalink required in order to use the volume
    modifier = DatasetObject(
        name="modifier",
        category="dishtowel",
        model="dtfspn",
        bounding_box=[0.34245, 0.46798, 0.07],
        visual_only=method_type
        == "Projection",  # Non-fluid adjacency requires the object to have collision geoms active
        abilities=abilities,
    )
    # Note: the following is a hacky trick done only for this specific demo that mutates the way the object applies particles;
    # the following trick should not be followed ever
    modifier._scene = env.scene
    modifier._scene_assigned = True
    modifier._created_manually = True
    modifier._prim = modifier._load()
    modifier_root_link_path = f"{modifier.prim_path}/base_link"
    if method_type == "Projection":
        metalink_path = f"{modifier.prim_path}/{modification_metalink[modifier_type]}"
        og.sim.stage.DefinePrim(metalink_path, "Xform")
        create_joint(
            prim_path=f"{modifier_root_link_path}/{modification_metalink[modifier_type]}_joint",
            body0=modifier_root_link_path,
            body1=metalink_path,
            joint_type="FixedJoint",
            enabled=True,
        )
    modifier._loaded = True
    modifier._post_load()
```

Reply:
Hmmm sorry, the original demo I wrote was already a bit hacky because we didn't have proper particle applier / remover objects in our dataset. I'm wondering whether we should just nix this and not allow thes alternative backdoor type methods? Esp. since we're just explicitly setting private variables which seems not very pythonic.

Are there any other usecases for "alternative methods"?

Reply:
There are lots - any kind of command / isaac API call / etc that creates sensors, particles, etc. that results in a pre-created prim that we just wrap in an XFormPrim for pose handling needs to still go through this procedure.